src/main/java/com/splunk/ibm/mq/metricscollector/MetricsPublisherJob.java

```diff
@@ -38,7 +38,9 @@ public void run() {
     } catch (Exception e) {
       logger.error("Error executing " + delegate.getClass().getSimpleName(), e);
     } finally {
-      latch.countDown();
+      if (latch != null) {
+        latch.countDown();
+      }
```

Comment:
We should avoid using nulls like this, where possible. It looks like you did this to account for the other case that you changed in the JobSubmitterContext to pass null. I'd say that we should not do this. If you need to have the job NOT count down the main job counter, then the JobSubmitterContext should pass a disposable (unrelated) countdown latch. 

Alternatively, and it's more work, but we could decouple the job from the latch implementation with a listener. Probably a separate PR if you wanted to go that route.

Reply:
I will follow your recommendation, thank you for the quick review.